kernel/kernel-api/src/main/java/io/delta/kernel/internal/ScanBuilderImpl.java

```diff
@@ -81,6 +83,22 @@ public ScanBuilder withReadSchema(TableClient tableClient, StructType readSchema
 
     @Override
     public Scan build() {
+        // TODO: support timestamp type partition columns
+        // Timestamp partition columns have complicated semantics related to timezones so block this
+        // for now
+        List<String> partitionCols = protocolAndMetadata.get()._2.getPartitionColumns();
```

Comment:
Wondering if we should do the check here or let it fail when we actually parse the partition value at the executor.

Reply:
I figured failing sooner rather than later was preferable but I can change it